ts/src/oxfun.ts

```diff
@@ -0,0 +1,2463 @@
+                    // {"success":false,"code":"20001","message":"timeframe is invalid"}
+                    // {"success":false,"code":"20001","message":"limit exceeds the maximum"}
+                    // {"success":false,"code":"20001","message":"limit is invalid"}
+                    // {"success":false,"code":"20001","message":"startTime/endTime is invalid"}
+                    // {"success":false,"code":"20001","message":"startTime and endTime must be within 7 days of each other"}
+                    // {"success":false,"code":"20001","message":"endTime must be greater than startTime"}
+                    // {"success":false,"code":"20001","message":"level exceeds the maximum"}
+                    // {"success":false,"code":"20001","message":"marketCode is invalid"}
+                    // {"success":false,"code":"30001","message":"Required parameter 'marketCode' is missing"}
+                    // {"event":null,"success":false,"message":"Validation failed","code":"0010","data":null} - failed transfer
+                    // {"success":false,"code":"20001","message":"subAcc is invalid"}
+                    // {"success":false,"message":null,"code":"500","timestamp":"2024-05-09T13:15:30.418+0000","data":null}
+                    // {"success":false,"code":"20001","message":"This market does not have a funding rate"}
+                    // {"success":false,"code":"30001","message":"Required parameter 'asset' is missing"}
+                    // {"success":false,"code":"20001","message":"startTime and endTime must be within 7 days of each other"}
+                    // {"success":false,"code":"30001","message":"Required parameter 'timestamp' is missing","data":null}
+                    // {"success":false,"code":"20001","message":"subAcc is invalid"}
+                    // {"success":false,"code":"30001","message":"Required parameter 'externalFee' is missing"}
+                    // {"success":false,"code":"35034","message":"Wallet API is not functioning properly, please try again or contact support."}
+                    // {"success":false,"code":"35046","message":"Error. Please refresh the page."}
+                    // {"success":false,"code":"30001","message":"Required parameter 'quantity' is missing"}
+                },
+                'broad': {
+                    // todo: add more error codes
+                },
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}): Promise<Market[]> {
+        /**
+         * @method
+         * @name oxfun#fetchMarkets
+         * @description retrieves data on all markets for bitmex
+         * @see https://docs.ox.fun/?json#get-v3-markets
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object[]} an array of objects representing market data
+         */
+        const responseFromMarkets = await this.publicGetV3Markets (params);
```

Comment:
If we need to make 2 different requests, we should do them "in parallel" using promise.all

Reply:
Done